repo/repository.go

```diff
@@ -137,6 +139,10 @@ type directRepository struct {
 	afterFlush []RepositoryWriterCallback
 }
 
+func (r *directRepository) CloseDebug(ctx context.Context) {
+	debug.StopProfileBuffers(ctx)
+}
```

Comment:
given how we have been calling `StartProfileBuffers`, I am not sure if you'll actually need to wrap `debug.StopProfileBuffers` in the repository object.